docs/spec/sources.md

```diff
@@ -362,5 +375,38 @@ _Structured Content Mode_ of the HTTP Protocol Binding for CloudEvents. Sources
 SHOULD send events to its
 [Destination](https://pkg.go.dev/github.com/knative/pkg/apis/duck/v1?tab=doc#Destination).
 
+### CloudEvent `knsource` extension
+
+Sources SHOULD produce CloudEvents with the `knsource` CloudEvent extension
+identifying the originating [Source Custom Object](#source-custom-objects). It
+is highly-recommended to produce the `knsource` CloudEvent extension.
```

Comment:
Isn't `SHOULD` read as "(highly) recommended"already ?

Reply:
yes, but you know, everyone can use a bit of extra motivation .... Anyway I'm happy to remove it.

Reply:
I removed.